src/server/zset_family.cc

```diff
@@ -2762,34 +2765,63 @@ void GeoSearchGeneric(ConnectionContext* cntx, const GeoShape& shape_ref, string
   // sort and trim by count
   SortIfNeeded(&ga, geo_ops.sorting, geo_ops.count);
 
-  // generate reply array withdist, withcoords, withhash
-  int record_size = 1;
-  if (geo_ops.withdist) {
-    record_size++;
-  }
-  if (geo_ops.withhash) {
-    record_size++;
-  }
-  if (geo_ops.withcoord) {
-    record_size++;
-  }
-  (*cntx)->StartArray(ga.size());
-  for (const auto& p : ga) {
-    // [member, dist, x, y, hash]
-    (*cntx)->StartArray(record_size);
-    (*cntx)->SendBulkString(p.member);
+  if (geo_ops.store == GeoStoreType::kNoStore){
+    // generate reply array withdist, withcoords, withhash
+    int record_size = 1;
     if (geo_ops.withdist) {
-      (*cntx)->SendDouble(p.dist / geo_ops.conversion);
+      record_size++;
     }
     if (geo_ops.withhash) {
-      (*cntx)->SendDouble(p.score);
+      record_size++;
     }
     if (geo_ops.withcoord) {
-      (*cntx)->StartArray(2);
-      (*cntx)->SendDouble(p.longitude);
-      (*cntx)->SendDouble(p.latitude);
+      record_size++;
+    }
+    (*cntx)->StartArray(ga.size());
+    for (const auto& p : ga) {
+      // [member, dist, x, y, hash]
+      (*cntx)->StartArray(record_size);
+      (*cntx)->SendBulkString(p.member);
+      if (geo_ops.withdist) {
+        (*cntx)->SendDouble(p.dist / geo_ops.conversion);
+      }
+      if (geo_ops.withhash) {
+        (*cntx)->SendDouble(p.score);
+      }
+      if (geo_ops.withcoord) {
+        (*cntx)->StartArray(2);
+        (*cntx)->SendDouble(p.longitude);
+        (*cntx)->SendDouble(p.latitude);
+      }
+    }
+  } else {
+    DCHECK(geo_ops.store == GeoStoreType::kStoreDist || geo_ops.store == GeoStoreType::kStoreHash);
+    ShardId dest_shard = Shard(store_key, ga.size());
+    AddResult add_result;
+    vector<ScoredMemberView> smvec;
+    for (const auto& p : ga) {
+      if (geo_ops.store == GeoStoreType::kStoreDist) {
+        smvec.emplace_back(p.dist / geo_ops.conversion, p.member);
+      } else {
+        DCHECK(geo_ops.store == GeoStoreType::kStoreHash);
+        smvec.emplace_back(p.score, p.member);
+      }
     }
+
+    auto store_cb = [&](Transaction* t, EngineShard* shard) {
+      if (shard->shard_id() == dest_shard) {
+        ZParams zparams;
+        zparams.override = true;
+        add_result = OpAdd(t->GetOpArgs(shard), zparams, store_key, ScoredMemberSpan{smvec}).value();
+      }
+      return OpStatus::OK;
+    };
+    cntx->transaction->Schedule();
+    cntx->transaction->Execute(std::move(store_cb), true);
```

Comment:
I'm not sure which one should I use here.  `schedule-execute` or `ScheduleSingleHopT`? It seems that I have to set "-4, 3, 3, 1" in `Register` to use  `schedule-execute`. But got error message:
_127.0.0.1:6379> GEORADIUSBYMEMBER Europe Madrid 700 KM WITHCOORD WITHDIST STORE test1
Error: Server closed the connection_

The search part works good. 

I'd appreciate it if anyone could give me some advice.

Reply:
Please do not add support for `STORE` and `STOREDIST` options. There are several reasons for this:
1. it requires some adjustments to transactional framework to support this non-standard command layout
2. this also requires advanced knowledge of how to build such commands. 
 
Having said that, implementing all the other options will  provide substantiation contribution and will make it easier for us to close the gaps.

Reply:
Thanks for your suggestion. I've updated my PR.
 
I'll be glad to make further changes to transactional framework if you could give me some suggestions.

Another thing on my to-do list is that GeoSearch fetches redundant data and subsequently filter them. To eliminate the intermediate result array, also requires some change to the standard search process which cannot take extra geo filter as input.